pkg/mining/miningManager.go

```diff
@@ -0,0 +1,138 @@
+	"fmt"
+	"os"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/tellor-io/telliot/pkg/config"
+	"github.com/tellor-io/telliot/pkg/contracts"
+	"github.com/tellor-io/telliot/pkg/db"
+	"github.com/tellor-io/telliot/pkg/logging"
+	"github.com/tellor-io/telliot/pkg/rpc"
+)
+
+type SolutionSink interface {
+	Submit(context.Context, *Result) (*types.Transaction, error)
+}
+
+const NumProcessors = 1
+
+func SetupMiningGroup(ctx context.Context, logger log.Logger, cfg *config.Config, contractInstance *contracts.ITellor) (*MiningGroup, error) {
+	var hashers []Hasher
+	level.Info(logger).Log("msg", "starting CPU mining", "threads", NumProcessors)
+	for i := 0; i < NumProcessors; i++ {
+		hashers = append(hashers, NewCpuMiner(int64(i)))
+	}
+	miningGrp, err := NewMiningGroup(ctx, logger, cfg, hashers, contractInstance)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating new mining group")
+	}
+	return miningGrp, nil
+}
+
+// MiningMgr manages mining, submiting a solution and requesting data.
+// In the tellor contract a solution is saved in slots where a value is valid only when it has 5 confirmed slots.
+// The manager tracks tx costs and profitThreshold is set it skips any transactions below the profit threshold.
+// The profit is calculated the same way as in the Tellor contract.
+// Transaction cost for submitting in each slot might be different so because of this
+// the manager needs to complete few transaction to gather the tx cost for each slot.
+type MiningMgr struct {
+	ctx              context.Context
+	close            context.CancelFunc
+	logger           log.Logger
+	ethClient        contracts.ETHClient
+	group            *MiningGroup
+	taskerCh         chan *Work
+	submitterCh      chan *Result
+	database         db.DataServerProxy
+	contractInstance *contracts.ITellor
+	cfg              *config.Config
+
+	toMineInput    chan *Work
+	solutionOutput chan *Result
+}
+
+// CreateMiningManager is the MiningMgr constructor.
+func CreateMiningManager(
+	logger log.Logger,
+	ctx context.Context,
+	cfg *config.Config,
+	database db.DataServerProxy,
+	contractInstance *contracts.ITellor,
+	taskerCh chan *Work,
+	submitterCh chan *Result,
+) (*MiningMgr, error) {
+
+	// Setup a MiningGroup for each account.
+	group, err := SetupMiningGroup(ctx, logger, cfg, contractInstance)
+	if err != nil {
+		return nil, errors.Wrap(err, "setup MiningGroup")
+	}
+
+	client, err := rpc.NewClient(logger, cfg, os.Getenv(config.NodeURLEnvName))
+	if err != nil {
+		return nil, errors.Wrap(err, "creating client")
+	}
+
+	//ops logging
+	logger, err = logging.ApplyFilter(*cfg, ComponentName, logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "apply filter logger")
+	}
+	ctx, close := context.WithCancel(ctx)
+	mng := &MiningMgr{
+		ctx:              ctx,
+		close:            close,
+		logger:           log.With(logger, "component", ComponentName),
```

Comment:
If we are going to run a submitter, miner and tasker for each pub key lets add it to the logs as well to know what is happening and for which account.
Add just the first 6 letters of the pub address, this will be anough to distinguish the logs.
```suggestion
		logger:           log.With(logger, "component", ComponentName,"pubKey",pubKey[:6]),
```

Reply:
I did the same for submitters.